From 5221bb6f7342164192d07eda0d26cc90d293c31d Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@gmail.com>
Date: Fri, 19 Apr 2019 09:50:15 +0200
Subject: [PATCH 10/79] events/core: annotate mutex_lock_double

To handle this warning:

kernel/events/core.c:11292:2: warning: releasing mutex 'dst_ctx->mutex' that was not held [-Wthread-safety-analysis]
        mutex_unlock(&dst_ctx->mutex);
        ^
kernel/events/core.c:11293:2: warning: releasing mutex 'src_ctx->mutex' that was not held [-Wthread-safety-analysis]
        mutex_unlock(&src_ctx->mutex);
        ^
---
 kernel/events/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/events/core.c b/kernel/events/core.c
index a1f112ecaf8f..ddd9fedb4bfe 100644
--- a/kernel/events/core.c
+++ b/kernel/events/core.c
@@ -10637,7 +10637,7 @@ perf_event_set_output(struct perf_event *event, struct perf_event *output_event)
 	return ret;
 }
 
-static void mutex_lock_double(struct mutex *a, struct mutex *b)
+static void mutex_lock_double(struct mutex *a, struct mutex *b) __attribute__((acquire_capability(a))) __attribute__((acquire_capability(b)))
 {
 	if (b < a)
 		swap(a, b);
-- 
2.17.1

